### ISSUE TITLE
'integerkey' switch in main open_db doesn't resolve sub-db names

### ISSUE THREAD

**Issue by madarez** [jnwatson/py-lmdb#251]
### Affected Operating Systems
* MacOS

### Affected py-lmdb Version
* 1.0.0

### py-lmdb Installation Method
* pip install lmdb

### Using bundled or distribution-provided LMDB library?

* Bundled
### Distribution name and LMDB library version
* (0, 9, 24)

### Machine "free -m" output

Memory: 5.47G of 16.38G free

```bash
vm_stat
Mach Virtual Memory Statistics: (page size of 4096 bytes)
Pages free:                                3610.
Pages active:                           1180342.
Pages inactive:                         1156361.
Pages speculative:                        23751.
Pages throttled:                              0.
Pages wired down:                       1006480.
Pages purgeable:                         109014.
"Translation faults":                1076335062.
Pages copy-on-write:                   41865122.
Pages zero filled:                    560193302.
Pages reactivated:                     75828700.
Pages purged:                          79323676.
File-backed pages:                       244541.
Anonymous pages:                        2115913.
Pages stored in compressor:             8871248.
Pages occupied by compressor:            823393.
Decompressions:                       118306660.
Compressions:                         156423612.
Pageins:                               42526337.
Pageouts:                                669951.
Swapins:                              160882156.
Swapouts:                             164856494.
```

### Other important machine info

N/A

### Describe Your Problem

I tried define integer-based keys for the main db in order to name integer-based names for sub-dbs. The lmdb sometimes works out without any issues, and it sometimes runs into segmentation faults. It's as if there're some information around integer-based keys when used for the name of the sub-dbs that I cannot locate in the docs.

### Errors/exceptions Encountered

Take the following snippet for example:

```python
import tempfile
import lmdb
import struct
fp = tempfile.TemporaryDirectory()
env = lmdb.Environment(path=fp.name, max_dbs=3)
pack_uint = lambda x: struct.pack('I', x)
db_name = pack_uint(7)
main_db = env.open_db(None , integerkey=True) # env.open_db(None)
db = env.open_db(db_name, integerkey=True)

with env.begin(db=db, write=True) as txn:
    txn.put(key=pack_uint(8), value=pack_uint(9), append=True, db=db)
    txn.put(key=pack_uint(88), value=pack_uint(99), append=True, db=db)

fp.cleanup()
```

Running the above snippet in a loop causes some of the python runs to fail, e.g.,
```bash
for i in {1..10}; do python3 -q -X faulthandler above_snippet.py; done
Fatal Python error: Segmentation fault

Current thread 0x000000010b6f95c0 (most recent call first):
  File "kk.py", line 13 in <module>
Segmentation fault: 11
Fatal Python error: Segmentation fault

Current thread 0x000000010f7a85c0 (most recent call first):
  File "kk.py", line 13 in <module>
Segmentation fault: 11
```
In the case above, the above snippet failed only twice out of the five run times. What I noticed as well was that all the failings happen on the second append and the first one always returns True.

Let's take a similar example:
```python
import tempfile
import lmdb
import struct
fp = tempfile.TemporaryDirectory()
env = lmdb.Environment(path=fp.name, max_dbs=6)
pack_item = lambda x: struct.pack('I', x)
db_name = pack_item(7)
db = env.open_db(db_name, integerkey=True)

def append(value):
  with env.begin(db=db, write=True) as txn:
    return txn.put(key=pack_item(txn.id()), value=pack_item(value), append=True, db=db)

assert append(8) == True
assert append(9) == True

main_db = env.open_db(None , integerkey=True) # env.open_db(None)
with env.begin(db=main_db, buffers=True) as txn:
  assert txn.stat(main_db)['entries'] == 1

db = env.open_db(db_name, integerkey=True)
with env.begin(db=db, buffers=True) as txn:
  assert txn.stat(db)['entries'] == 2

fp.cleanup()
```
In this snippet, I don't encounter a segmentation fault, but the second assert fails to see the appended 8. However, after inspecting it, in the failed cases `txn.stat(db)` is empty whose 'entries' is actually zero. This time around, this case fails regardless of the second append in the second snippet above.

Now, if you change `main_db = env.open_db(None)` in either snippet, there will be no problem anymore. That's what caused me to believe that this has something to do with the `integerkey` switch.

### Describe What You Expected To Happen

I expected both snippets to succeed consistently across multiple runs.
### Describe What Happened Instead

The Python process crashed (Segmentation fault: 11) in the first snippet and the second assertion fails in the second example.

**Comment by jnwatson** [comment]
There's a bug here, but it is that I don't verify that there are no flags with the default DB. From the docs:

> The `dupsort`, `integerkey`, `integerdup`, and `dupfixed` parameters are
> ignored if the database already exists.  The state of those settings are
> persistent and immutable per database.  See :py:meth:`_Database.flags`
> to view the state of those options for an opened database.  A consequence
> of the immutability of these flags is that the default non-named database
> will never have these flags set.

**Comment by jnwatson** [comment]
Resolved by 6aeba4ccb
